src/video/cocoa/cocoa_wnd.mm

```diff
@@ -261,6 +261,14 @@ - (void)unregisterObserver
 {
 	[ [ NSNotificationCenter defaultCenter ] removeObserver:self ];
 }
+
+/**
+ * Indicate to AppKit that we don't interfere with its secure coding.
+ */
+- (BOOL)applicationSupportsSecureRestorableState
+{
+	return YES;
+}
```

Comment:
This function (and its comment) are very vague and non-descriptive. The motivation also doesn't mention why it should exist. This means it is not unlikely someone removes it somewhere in the future, as it "seems to have no effect".

Would you mind given a more clear indication why this is needed, and what it is doing? Either in the motivation of the PR, or if possible, in the comment?

When I read the documentation, it seems unrelated to this PR, and has nothing to do with the social plugins? But I might be misunderstanding the very few words Apple's documentation uses to describe this function. Hence my remark here :)

Reply:
Since macOS 14 Sonoma, macOS logs a warning about not explicitly stating whether secure state recording is supported by the app.
By default, it logs a warning and returns true.
This code does the same without the warning, and since it's so small and changes nothing of value (OpenTTD encodes no state of its own to restore from to begin with), it felt more fitting to include here than to make a separate PR.

Reply:
Please don't sneak other change in PRs; it makes a nightmare for us when something breaks because of it. When you do a git blame on this line, you will find this PR, with no mention in the description about it what so ever. That would be incredibly hard to figure out why it was added and what it tried to solve.

So yeah, even for these small things, please just create a tiny small PR :) Future-us will thank you.